### PR TITLE
gha: Also run k8s tests on qemu-sev

### PR DESCRIPTION
**Pull request by ryansavino**
Adding the k8s tests for qemu-sev

Fixes: #6711

### CODE REVIEW

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-sev.yaml:1]: This workflow is almost identical to run-k8s-tests-on-tdx.yaml, except for the vmm, runner node and perhaps the kubeconfig (I explain later).

So one possibility is to combine the TDX and SEV (possible others TEEs on future) into a single workflow. Something like:
 * Rename to run-k8s-tests-on-tdx.yaml -> run-k8s-tests-on-tee.yaml
 * Change the workflow:
```
diff --git a/.github/workflows/run-k8s-tests-on-tee.yaml b/.github/workflows/run-k8s-tests-on-tee.yaml
index 1777a16c8..d497a6667 100644
--- a/.github/workflows/run-k8s-tests-on-tee.yaml
+++ b/.github/workflows/run-k8s-tests-on-tee.yaml
@@ -1,4 +1,4 @@
-name: CI | Run kubernetes tests on TDX
+name: CI | Run kubernetes tests on TEE
 on:
   workflow_call:
     inputs:
@@ -19,9 +19,17 @@ jobs:
       matrix:
         vmm:
           - qemu-tdx
-    runs-on: tdx
+          - qemu-sev
+        include:
+          - vmm: qemu-tdx
+            runner: tdx
+            kubeconfig: /etc/rancher/k3s/k3s.yaml
+          - vmm: qemu-sev
+            runner: sev 
+            kubeconfig: /path/to/kubeconfig
+    runs-on: ${{ matrix.runner }}
     env:
-      KUBECONFIG: /etc/rancher/k3s/k3s.yaml
+      KUBECONFIG: ${{ matrix.kubeconfig }}
     steps:
```

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-sev.yaml:22]: @ryansavino do we have already the SEV machine added on github as a self-hosted runner?

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-sev.yaml:24]: IIUC @fidencio has installed Rancher on TDX machine already, and the idea is to not re-install k8s over and over again for each job. We probably should be doing the same thing for SEV, but perhaps install a vanilla Kubernetes (with kubeadm?)? Unless, @ryansavino , you don't care about what k8s flavor to use :)

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (ryansavino) [reply .github/workflows/run-k8s-tests-on-sev.yaml:1]: This could work. My only concern is that sev and snp will need to add additional bats files to the K8S_TEST_UNION, but we can accommodate in the tee yaml.

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (ryansavino) [reply .github/workflows/run-k8s-tests-on-sev.yaml:22]: Yep, both a rome and a milan have been added for sev, sev-es and sev-snp respectively.

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (ryansavino) [reply .github/workflows/run-k8s-tests-on-sev.yaml:24]: I installed a vanilla kubernetes instance via kubeadm, but I can change it to rancher if that is required.

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (wainersm) [reply .github/workflows/run-k8s-tests-on-sev.yaml:1]: I talked with @ryansavino and we decided to not merge the worflows for now. However, I will be opening an issue for the consolidation, that we work to do when the pipelines are all merged and stable.

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-sev.yaml:39]: There should be added an entry for kata-qemu-sev on the kata-runtimeClasses.yaml file. Instead of doing that on this PR, it should be done on https://github.com/kata-containers/kata-containers/pull/6591 as it already enable the kata configuration.toml for sev.

**Review by wainersm — APPROVED** [review]
Thanks @ryansavino , it is a good start. Once we have this landed in main, we will be able to fully test the PRs that add the sev/snp artifacts.

**Comment by wainersm** [comment]
run-k8s-tests (clh) and run-k8s-tests (dragonball)  failures aren't related with the changes on this PR. Once we have the 2 required approvals it can be merged.

**Review by sameo — APPROVED** [review]

File: .github/workflows/run-k8s-tests-on-sev.yaml
Comment (fidencio) [reply .github/workflows/run-k8s-tests-on-sev.yaml:1]: As discussed a few times in the past, we're hoping that the SEV / SNP additional bats tests will be generic enough to accommodate other TEEs as well (with internal implementations being different), thus I do believe it does make sense to merge the files.

Comment (wainersm) [reply .github/workflows/run-k8s-tests-on-sev.yaml:1]: Yes, refactoring the sev/snp tests will be a requirement. But IIUC at this point we are running the generic k8s tests, none related with TEE. Also @ryansavino has started the work with refactoring the sev tests.

**Pull request merged**
